Sming/SmingCore/Network/MqttClient.cpp

```diff
@@ -9,6 +9,19 @@
 
 #include "../SmingCore.h"
 
+MqttClient::MqttClient()
+	: TcpClient((bool)false),
```

Comment:
I would prefer to allow destruct to be controlled, if needed. Make the MqttClient constructor to have one optional parameter `bool autoDestruct=false` and then pass the `autoDestruct` to TcpClient(autoDestruct)